nixos/modules/services/misc/rkvm.nix

```diff
@@ -0,0 +1,168 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+let
+  cfg = config.services.rkvm;
+  toml = pkgs.formats.toml { };
+in
+{
+  meta.maintainers = with maintainers; [ ckie ];
+
+  options.services.rkvm = {
+    enable = mkOption {
+      default = cfg.server.enable || cfg.client.enable;
+      defaultText = literalExpression "cfg.server.enable || cfg.client.enable";
+      type = types.bool;
+      description = lib.mdDoc ''
+        Whether to enable rkvm, a Virtual KVM switch for Linux machines.
+      '';
+    };
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.rkvm;
+      defaultText = literalExpression "pkgs.rkvm";
+      description = mdDoc ''
+        The rkvm derivation to use.
+      '';
+    };
+
+    server = {
+      enable = mkEnableOption "rkvm server daemon (input transmitter)";
+
+      settings = mkOption {
+        type = types.submodule
+          {
+            freeformType = toml.type;
+            options = {
+              listen = mkOption {
+                type = types.str;
+                default = "0.0.0.0:5258";
+                description = mdDoc ''
+                  Network listen address including a port. Leave blank to listen on all available network interfaces.
+                  On *NIX you can specify an absolute path to listen on a UNIX domain socket.
+                '';
+              };
+
+              switch-keys = mkOption {
+                type = types.listOf types.str;
+                default = [ "left-alt" "right-alt" ];
+                description = mdDoc ''
+                  A key list specifying a host switch combination.
+
+                  _A list of key names is available in <https://github.com/htrefil/rkvm/blob/master/switch-keys.md>._
+                '';
+              };
+
+              certificate = mkOption {
+                type = types.path;
+                default = "/etc/rkvm/certificate.pem";
```

Comment:
I'll have a look to see if I can find the details.

As for autogenerating, I think it might bring a bit of extra complexity. The certificates requires you to specify the ip address (or a list of them) that the server will listen on, and that doesn't seem to go hand in hand with the `0.0.0.0` default. If we make a list option for this, should the certificate be regenerated if it gets changed?